### ISSUE TITLE
Invalid name: "@sailshq/lodash"

### ISSUE THREAD

**Issue by CuAnnan** [balderdashy/sails#3984]
**Sails version**: current
**Node version**: v6.9.5
**NPM version**: v3.10.10
**Operating system**: OSX 10.12.2

<hr/>

sudo npm install sails -g results in Invalid name: "@sailshq/lodash"

[npm-debug.txt](https://github.com/balderdashy/sails/files/773579/npm-debug.txt)

**Comment by mikermcneil** [comment]
@CuAnnan I tried replicating with your environment more or less exactly, but I can't recreate the issue on my end:
![image](https://cloud.githubusercontent.com/assets/618009/22942082/9f160e9a-f2ad-11e6-99af-cac5af891ff2.png)

Also tried:
![image](https://cloud.githubusercontent.com/assets/618009/22941694/2655fcbe-f2ac-11e6-8b3b-53291ed3b0ee.png)

This seems like maybe an issue that sometimes occurs with scoped modules on npm ([related?](https://github.com/npm/npm/issues/8927))

You might try running `npm cache clear` -- and also try `npm install sails -g` (without the `sudo`).  Would you let me know if that works?

**Comment by CuAnnan** [comment]
Running without sudo, I got an EACCESS: permission denied, access '/usr/local/lib/node_modules"

After rerunning npm cache clear I got a host of new errors when I ran the install with sudo again.

npm Err! unpack Could not read data for @sailshq/lodash@3.10.2
npm Err! unpack Could not read data for @mapbox/geojsonhint@1.2.1

Which returns me to the Invalid name issue.

**Comment by CuAnnan** [comment]
I've also tried setting the prefix for npm to ~/.npm-global, so as to avoid the sudo, but to no avail.
Exactly the same output.

I'm going to try nuking node and using homebrew's node instead of nodejs.org; as I think this may actually be a problem with how node is installed.

**Comment by CuAnnan** [comment]
Fixed all of my problems. Thank you for your help, sorry to have wasted your time.

**Comment by mwkaicz** [comment]
I have same problem:

```npm ERR! Linux 3.2.0-4-amd64
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install"
npm ERR! node v5.2.0
npm ERR! npm  v3.3.12

npm ERR! Invalid name: "@sailshq/lodash"
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /srv/node/Chart3D/npm-debug.log
```

----------------------------
```
root@craftengine:/srv/node/Chart3D# npm install sails -g
npm WARN install Couldn't install optional dependency: Invalid name: "@sailshq/lodash"
```
what's the solution?

**Comment by CuAnnan** [comment]
I nuked the install of Node that I got from their site altogether, [see this article](http://benznext.com/completely-uninstall-node-js-from-mac-os-x/).

Then I used homebrew to install it.

I am lead to believe the issue has to do with permissions.  
But I could not figure out what the right permissions were with the node distro, and homebrew just took care of it for me.  
I don't know if that will help in your particulars.
